controllers/update.go

```diff
@@ -70,24 +70,75 @@ func (r *ComponentReconciler) updateComponentDevfileModel(req ctrl.Request, hasC
 	for _, kubernetesComponent := range kubernetesComponents {
 		compUpdateRequired := false
 		// Update for Replica
-		currentReplica := 0
+		currentReplica := 1 // default value is 1
+		keyFound := true
+
 		if len(kubernetesComponent.Attributes) == 0 {
 			kubernetesComponent.Attributes = attributes.Attributes{}
+			keyFound = false
 		} else {
 			var err error
 			currentReplica = int(kubernetesComponent.Attributes.GetNumber(devfile.ReplicaKey, &err))
 			if err != nil {
 				if _, ok := err.(*attributes.KeyNotFoundError); !ok {
 					return err
+				} else {
+					keyFound = false
+					currentReplica = 1 //if an error is raised, it'll set currentReplica to 0 so we need to reset back to the default
 				}
 			}
 		}
 
-		numReplicas := util.GetIntValue(component.Spec.Replicas)
-		if currentReplica != numReplicas {
-			log.Info(fmt.Sprintf("setting devfile component %s attribute component.Spec.Replicas to %v", kubernetesComponent.Name, numReplicas))
-			kubernetesComponent.Attributes = kubernetesComponent.Attributes.PutInteger(devfile.ReplicaKey, numReplicas)
-			compUpdateRequired = true
+		numReplicas := 1 //default value
+		if component.Spec.Replicas != nil {
+			numReplicas = util.GetIntValue(component.Spec.Replicas)
+			// Component.Spec.Replicas will override any other settings.
+			// We will write the attribute if it doesn't exist for the initial creation case when comp.spec.replica is 1
+			if currentReplica != numReplicas || !keyFound {
+				log.Info(fmt.Sprintf("setting devfile component %s attribute component.Spec.Replicas to %v", kubernetesComponent.Name, numReplicas))
+				kubernetesComponent.Attributes = kubernetesComponent.Attributes.PutInteger(devfile.ReplicaKey, numReplicas)
+				compUpdateRequired = true
+			}
+		} else {
+			//check to see if we have an inlined deployment
+			isDeployReplicaSet := false
+			inlined := kubernetesComponent.Kubernetes.Inlined
+			if inlined != "" {
+				log.Info(fmt.Sprintf("reading the kubernetes inline from component %s", component.Name))
+				src := parser.YamlSrc{
+					Data: []byte(inlined),
+				}
+
+				values, err := parser.ReadKubernetesYaml(src, nil, nil)
+				if err != nil {
+					return err
+				}
+
+				resources, err := parser.ParseKubernetesYaml(values)
+				if err != nil {
+					return err
+				}
+
+				if len(resources.Deployments) > 0 {
+					replica := resources.Deployments[0].Spec.Replicas
+					if replica != nil {
+						isDeployReplicaSet = true
+						//remove the deployment/replicas attribute which can be left behind if we go from a set value to an unset value
+						if kubernetesComponent.Attributes.Exists(devfile.ReplicaKey) {
+							num := int(kubernetesComponent.Attributes.GetNumber(devfile.ReplicaKey, nil))
```

Comment:
```suggestion
							num := int(kubernetesComponent.Attributes.GetNumber(devfile.ReplicaKey, &err))
```
I dont think we should ignore the err

Reply:
I assumed no errors at this point if key is found.  I guess there may be other errors